x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/ranges/range_editor.tsx

```diff
@@ -0,0 +1,412 @@
+    defaultMessage: 'Granularity',
+  });
+
+  return (
+    <>
+      <EuiFormRow
+        label={sectionLabel}
+        data-test-subj="indexPattern-ranges-section-label"
+        labelAppend={
+          <EuiSwitch
+            label={i18n.translate('xpack.lens.indexPattern.ranges.autoInterval', {
+              defaultMessage: 'Auto interval',
+            })}
+            checked={autoIntervalEnabled}
+            onChange={(e) => toggleAutoInterval(e.target.checked)}
+            data-test-subj="indexPattern-ranges-auto-interval"
+            compressed
+          />
+        }
+      >
+        <>
+          <EuiFlexGroup gutterSize="s" responsive={false} wrap>
+            <EuiFlexItem>
+              <EuiFieldNumber
+                data-test-subj="lens-range-interval-field"
+                value={autoIntervalEnabled ? '' : interval}
+                onChange={({ target }) => onIntervalChange(Number(target.value))}
+                prepend={i18n.translate('xpack.lens.indexPattern.ranges.min', {
+                  defaultMessage: 'Min Interval',
+                })}
+                min={0}
+                placeholder={autoIntervalEnabled ? 'Auto' : ''}
+                disabled={autoIntervalEnabled}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+
+          <EuiButtonEmpty iconType="controlsHorizontal" onClick={() => onToggleEditor()}>
+            {i18n.translate('xpack.lens.indexPattern.ranges.customIntervalsToggle', {
+              defaultMessage: 'Create custom intervals',
+            })}
+          </EuiButtonEmpty>
+        </>
+      </EuiFormRow>
+    </>
+  );
+};
+
+const RangePopover = ({
+  range,
+  setRange,
+  Button,
+}: {
+  range: RangeType;
+  setRange: (newRange: RangeType) => void;
+  Button: React.FunctionComponent<{ onClick: MouseEventHandler }>;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [tempRange, setTempRange] = useState(range);
+  const { from, to } = tempRange;
+
+  const safeLabel = getSafeLabel(tempRange);
+
+  const gteAppendLabel = i18n.translate('xpack.lens.indexPattern.ranges.greaterThanOrEqualAppend', {
+    defaultMessage: '\u2264',
+  });
+  const gteTooltipContent = i18n.translate(
+    'xpack.lens.indexPattern.ranges.greaterThanOrEqualTooltip',
+    {
+      defaultMessage: 'Greater than or equal to',
+    }
+  );
+  const ltPrependLabel = i18n.translate('xpack.lens.indexPattern.ranges.lessThanPrepend', {
+    defaultMessage: '\u003c',
+  });
+  const ltTooltipContent = i18n.translate('xpack.lens.indexPattern.ranges.lessThanTooltip', {
+    defaultMessage: 'Less than',
+  });
+
+  return (
+    <EuiPopover
+      ownFocus
+      isOpen={isPopoverOpen}
+      closePopover={() => {
+        setIsPopoverOpen(false);
+        if (isValidRange(tempRange)) {
+          // reset the temporary range for later use
+          setTempRange(range);
+          // send the range back to the main state
+          setRange(tempRange);
```

Comment:
@dej611 in my filters PR we've decided that to make the behaviour more consistent to lens, we'll update and save state on change and not on blur/submit. When you have a moment, please take a look at my PR and let me know your thoughts.